src/btree/bt_read.c

```diff
@@ -250,8 +250,8 @@ __wt_page_in_func(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags
             if (LF_ISSET(WT_READ_CACHE))
                 return (WT_NOTFOUND);
 
-            /* Optionally limit reads to internal pages only. */
-            if (LF_ISSET(WT_READ_CACHE_LEAF) && F_ISSET(ref, WT_REF_FLAG_LEAF))
+            /* Optionally limit reads to internal pages and leaf pages with overflow keys only. */
+            if (LF_ISSET(WT_READ_CACHE_LEAF_NO) && F_ISSET(ref, WT_REF_FLAG_LEAF))
```

Comment:
Since we now have a custom skip function for the walk, do we still need the `WT_READ_CACHE_LEAF_NO` flag? Can't we skip those pages in our custom skip function?

Reply:
During the implementation, I first removed that flag, but there are other flags that must be passed as part of the checkpoint. Otherwise, an assert is getting hit in `__wt_page_in_func`.  The other flag is WT_READ_NO_WAIT.

Currently, we don’t set WT_READ_NO_WAIT for checkpoint flow. If we remove the WT_READ_CACHE_LEAF_NO flag, we must set that also to let the assert get passed in `__wt_page_in_func`. I doubt that WT_READ_NO_WAIT flag usage as part of the tree walk may cause problems to the checkpoint. So I retained the WT_READ_CACHE_LEAF_NO flag. I will create another ticket to remove the WT_READ_CACHE_LEAF_NO flag after analysing the WT_READ_NO_WAIT flag further.

Reply:
Agreed -- I'm pretty sure it's wrong as written, but happy to fix it later.

Reply:
Removed the `WT_READ_CACHE_LEAF_NO` macro as it is not required with custom skip function. The custom skip function takes care of reading the leaf pages with overflow keys also if required.